### ISSUE TITLE
expo-localization throw exception java.lang.NullPointerException: null cannot be cast to non-null type kotlin.Any on Android

### ISSUE THREAD

**Issue by cnhongwei** [expo/expo#16509]
### Summary

I use expo sdk 44 with expo-localization.
On some Android phones, Localization.locale will throw an exception "java.lang.NullPointerException: null cannot be cast to non-null type kotlin.Any"
I check the code,  in class LocalizationModule
```java
  override fun getConstants(): Map<String, Any> {
    val constants = HashMap<String, Any>()
    val bundle = bundledConstants
    for (key in bundle.keySet()) {
      constants[key] = bundle[key] as Any
    }
    return constants
  }
```
when the key is 'region', then value is null.  so throw the exception.

maybe change Any to Any?

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

44

### Environment

  expo-env-info 1.0.2 environment info:
    System:
      OS: macOS 12.2
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 12.22.8 - ~/.nvm/versions/node/v12.22.8/bin/node
      npm: 6.14.15 - ~/.nvm/versions/node/v12.22.8/bin/npm
      Watchman: 2022.02.14.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.11.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.2, iOS 15.2, macOS 12.1, tvOS 15.2, watchOS 8.3
      Android SDK:
        API Levels: 28, 29, 30
        Build Tools: 28.0.3, 29.0.0, 29.0.1, 29.0.2, 29.0.3, 30.0.3
        System Images: android-21 | Google APIs Intel x86 Atom, android-24 | Google Play Intel x86 Atom, android-26 | Google Play Intel x86 Atom, android-27 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom, android-30 | Google Play Intel x86 Atom_64
        Android NDK: 22.1.7171670
    IDEs:
      Android Studio: 2021.1 AI-211.7628.21.2111.8193401
      Xcode: 13.2.1/13C100 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^44.0.0 => 44.0.1
      react: 17.0.1 => 17.0.1
      react-dom: 17.0.1 => 17.0.1
      react-native: 0.64.3 => 0.64.3
      react-native-web: 0.17.1 => 0.17.1
    npmGlobalPackages:
      expo-cli: 5.2.0
    Expo Workflow: bare

### Reproducible demo

import * as Localization from 'expo-localization';

Localization.locale;

**Comment by barthap** [comment]
> when the key is 'region', then value is null. so throw the exception.
>
> maybe change Any to Any?

You're right, chanigng `Any` to `Any?` should fix the issue. You can open a PR if you'd like to 😁

**Comment by GaelCO** [comment]
Hi, 

Our application has the same problem for some devices  (Galaxy S9).
This issue seems already fixed by nomi9995 in the main branch : https://github.com/expo/expo/commit/98aa7941746da9e62a6f280686672f3de59c76fd

Do you know when this fix will be delivered ?

**Comment by barthap** [comment]
For fully managed workflow (Expo Go) - with SDK 45 (April/May)
For bare/prebuild workflow - sometimes packages with bug fixes are published more frequently, but I cannot promise anything.

As the issue is fixed already in 98aa794, closing it now.
